book/integration-service.md

```diff
@@ -0,0 +1,271 @@
+
+- When a build pipeline completes, it creates an Snapshot CR representing a new collection of components that should be tested together.
+- When it sees a new Snapshot CR (created either by itself or by a user), it coordinates deployment of the application for testing. It does this by creating a new “ephemeral” StoneSoup Environment CRs which should in turn trigger the GitOps Service to provision environment (and a sub-workspace in CPS)
+- When it sees the environment is ready, it triggers Tekton PipelineRuns supplied by the user that run application tests and validation against the app running in that ephemeral Environment.
+- When it sees that all of the test PipelineRuns that it created have passed, it finishes its work by triggering a deployment of the application to the lowest known “static” environment defined by the user, usually staging.
+- Finally, if any automatic ReleasePlans have been created by the user in the workspace, it will create a Release CR signalling intent to release the tested content, to be carried out by the release-service (separate threat model).
+
+
+### Goals
+
+- We need to be able to deploy a specific set of images to an environment and record that result somewhere.
+- Given a specific set of images, you can tell if they passed application validation tests or not
+- We must have some way of tracking a specific set of images where we can update one component image at a time in order to support 'slow loop' application validation testing.
+- To be SLSA compliant, we should also be able to tell what version of the build and test pipelines were used to create each component image.
+
+The diagram below shows the interaction of the integration service and other services.
+
+![](../diagrams/hacbs-data-flow.jpg)
+
+
+### Dependencies
+
+The [Integration Service](./integration-service.md) is dependent on the following services:
+- [Pipeline Service](./pipeline-service.md)
+  - Pipeline execution, Pipeline logging
+- [GitOps Service](./gitops-service.md)
+  - Provides the facility to create
+    - Snapshots used to test Builds
+    - Environment to test the Application on
+    - SnapshotEnvironmentBindings to have the Snapshot deployed to a specific Environment
+- [Hybrid Application Service](./hybrid-application-service.md)
+  - Provides the Application and Component model. Integration Service updates the pullspec reference on the Component CR when a component passes its required testing.
+- [Release Service](./release-service.md)
+  - Provides the ReleasePlan that will be used to determine if integration-service should create a Release
+- [Enterprise Contract Service](./enterprise-contract.md)
+  - Provides facilities to validate whether content has passed the Enterprise Contract.
+
+
+## Terminology
+
+**Prelim Pipeline** - This is the test pipeline run for the Component of an Application that gets triggered by a completed Build pipeline. Deprecated name, refer to the Component Pipeline
```

Comment:
A suggestion: maybe reorganize this into a `## Terminology` section that contains only current terminology, but then add a section just below it titled `### Outdated Terminology` that mentions the old Prelim and Final terms.

(As a reader - I gotmomentarilly confused when the _first_ term I encountered was deprecated.)

Reply:
I feel like I would go further. Since deprecated names are only really used in the early days, I would imagine that they would be most likely to only be looked up by searching. I think we would be able to remove them from the terminology and just indicate under the replacement terminology what other names it has been called in the past. If no references to these deprecated names can be found, I wouldn't include them in the terminology.

```
**Component Pipeline** - This is the test pipeline run for the Component of an Application that gets triggered by a completed Build pipeline. This is the replacement for the deprecated **Prelim Pipeline** which should no longer be used.
```